### PR TITLE
Update links to use channel library

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: router/xlink_transport/bind.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package xlink_transport
```
Comment (dovholuknf) [inline router/xlink_transport/bind.go:1]: header comments

File: router/handler_link/bind.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package handler_link
```
Comment (dovholuknf) [inline router/handler_link/bind.go:1]: header comments

File: router/handler_link/bind.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	binding.AddErrorHandler(newErrorHandler(self.xlink, self.ctrl))
+	binding.AddTypedReceiveHandler(newPayloadHandler(self.xlink, self.forwarder))
+	binding.AddTypedReceiveHandler(newQueuingAckHandler(self.xlink, self.forwarder, closeNotify))
+	binding.AddTypedReceiveHandler(&latency.LatencyHandler{})
+	binding.AddTypedReceiveHandler(newControlHandler(self.xlink, self.forwarder, closeNotify))
+	binding.AddPeekHandler(metrics2.NewChannelPeekHandler(self.xlink.Id().Token, self.forwarder.MetricsRegistry()))
+	binding.AddPeekHandler(trace.NewChannelPeekHandler(self.xlink.Id().Token, ch, self.forwarder.TraceController(), trace.NewChannelSink(self.ctrl.Channel())))
+
+	if self.trackLatency {
+		latencyMetric := self.metricsRegistry.Histogram("link." + self.xlink.Id().Token + ".latency")
```
Comment (dovholuknf) [inline router/handler_link/bind.go:76]: is there any way to embed `self.xlink.Id().Token` as a tag or anything else? aren't unbounded names like this 'frowned on'?

File: router/handler_link/bind.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		go latency.ProbeLatencyConfigurable(config)
+	}
+
+	return nil
+}
+
+func (self *bindHandler) verifyLink(l xlink.Xlink, ch channel.Channel) error {
+	var fingerprints []string
+	for _, cert := range ch.Certificates() {
+		fingerprint := fmt.Sprintf("%x", sha1.Sum(cert.Raw))
```
Comment (dovholuknf) [inline router/handler_link/bind.go:102]: you'd think this was just some function. does it make sense to put somewhere that is reusable?

File: router/handler_link/bind.go (hunk 3 above)
Comment (plorenz) [reply router/handler_link/bind.go:76]: Metric names have to be unique. This gets transformed when it goes through the events subsystem into a metric named `link.latency` with tags for the source and target routers.

**Review by dovholuknf — APPROVED** [review]
lgtm. few comments here/there

File: router/handler_link/bind.go (hunk 4 above)
Comment (plorenz) [reply router/handler_link/bind.go:102]: I checked and there is an nfpem.FingerprintFromX509. I'll use that

File: router/xlink_transport/bind.go (hunk 1 above)
Comment (plorenz) [reply router/xlink_transport/bind.go:17]: fixed

File: router/handler_link/bind.go (hunk 2 above)
Comment (plorenz) [reply router/handler_link/bind.go:1]: fixed

File: router/handler_link/bind.go (hunk 4 above)
Comment (plorenz) [reply router/handler_link/bind.go]: fixed

**Pull request merged**